### ISSUE TITLE
[Code] Enable gradient in renderer.py

### ISSUE THREAD

**Issue by NK-CS-ZZL** [Haian-Jin/TensoIR#12]
An amazing work and congrats! I have a question in the code and hope for a reply if you're convenient. In the file `renderer.py` Line 37, the TensoIR model is warped by `torch.enable_grad()`, so I wonder why we need the gradient here?

**Comment by Haian-Jin** [comment]
Hi,

This is because the code warped will compute derived normals as done [here](https://github.com/Haian-Jin/TensoIR/blob/main/models/tensorBase_rotated_lights.py#L840). The process of computing derived normals will call `torch.autograd.grad`, for which gradient computing is necessary. Since I warp the evaluation code with `torch.no_grad` (such as [here](https://github.com/Haian-Jin/TensoIR/blob/main/renderer.py#L135) ), an error will arise when evaluating if I don't warp the code you mentioned with `torch.enable_grad()` because there will be no gradient by default when evaluating.

Hope my answer can solve your problem.

**Comment by NK-CS-ZZL** [comment]
Since the function `compute_derived_normals` has been decorated by `@torch.enable_grad()`, we do have the gradient in this func whether we use `torch.enable_grad` or `torch.no_grad` outside it. 
Besides, using `torch.no_grad` prevents the gradient accumulation outside the func, which is more vram-friendly. 
In my environment, these two settings provide the same results, but using `torch.no_grad` leads to less memory consumption.

**Comment by Haian-Jin** [comment]
You are right. The code you mentioned is unnecessary. I will delete it later.
Thanks for your suggestion!

**Comment by NK-CS-ZZL** [comment]
You're welcome and thanks for the reply.
